File: playground/javascript/queue.js
class Node {
  constructor(value, next) {
    this.value = value;
    this.next = next;
  }
}

class Queue {
  constructor() {
    this.front = null;
  }
  push(value) {
    // 큐가 비있을때
    if (!this.front) {
      let newNode = new Node(value, null);
      this.front = newNode;
      return;
    }

    // 큐에 값있을때
    if (this.front) {
      let currNode = this.front;

      while (currNode.next !== null) {
        currNode = currNode.next;
      }
      currNode.next = new Node(value, null);
    }
  }
  pop() {
    if (!this.front) {
      return;
    }
    let currFrontNode = this.front;
    this.front = currFrontNode.next;

    return currFrontNode.value;
  }
  isEmpty() {
    return this.front === null;
  }
}

const queue = new Queue();

queue.push(1);
queue.push(2);
console.log(queue.isEmpty());
console.log(queue.pop());
console.log(queue.pop());
console.log(queue.pop());

File: playground/dataStructure/queue.py
class Node:
    def __init__(self, value):
        self.value = value
        self.next = None


class Queue:
    def __init__(self):
        self.head = None
        self.tail = None

    def enqueue(self, value):
        pass
        # new_node = Node(value)
        #
        # if self.head is None:
        #     self.head = new_node
        #     self.tail = new_node
        # else:
        #     self.tail.next = new_node
        #     self.tail = self.tail.next

    def dequeue(self):
        pass
        # if self.head is None:
        #     return -1
        #
        # front = self.head.data
        # self.head = self.head.next
        #
        # if self.head is None:
        #     self.tail = None
        # return front
